CMakeLists.txt

```diff
@@ -501,6 +501,17 @@ endif()
 # Trezor support check
 include(CheckTrezor)
 
+# As of OpenBSD 6.8, -march=<anything> breaks the build
+function(set_default_arch)
+  if (OPENBSD)
+    set(ARCH default)
+  else()
+    set(ARCH native)
+  endif()
+
+  set(ARCH ${ARCH} CACHE STRING "CPU to build for: -march value or 'default' to not pass -march at all")
```

Comment:
Are you intending to set ARCH twice, with different config here ? Or is the first one meant to be a local ?

Reply:
The first one is local to the function, it's why I added the function in the first place.

Reply:
How do you tell ?

Reply:
It's how cmake works, variables are scoped to the enclosing function or, if not in a function, the current directory. IOW `function` and `add_subdirectory` are the only way to create a new scope. https://cmake.org/cmake/help/latest/command/set.html

Reply:
Why is the first set local to the function and the second is not? cmake always confuses me.

Reply:
The second one uses the `CACHE` option to the `set` command and so sets a cache variable, which is a totally different thing altogether. (Yes, cmake is, um, special.) These are meant to be user-settable (e.g. with the `cmake -Dx=y` option), persist across runs, and have project-wide scope. See also https://cmake.org/cmake/help/latest/manual/cmake-language.7.html#cmake-language-variables

Reply:
Thanks